Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Data.Repositories;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly TimeSpan ErrorRemoveDelay = TimeSpan.FromSeconds(5);
+
+        private readonly DateTime _utcNow = DateTime.UtcNow;
```

Comment:
This should be pulled from an injected `ISystemClock` service. Although, I'm not sure if MODiX actually has that setup, or if I'm just thinking of my own projects. If not, you should add it.

Reply:
I dont think modix currently does